tests/infra.py

```diff
@@ -79,3 +84,131 @@ def __getattr__(self, item: str) -> typing.Any:
             return super(DSSAsserts, self).__getattr__(item)  # type: ignore
         else:
             raise AttributeError(item)
+
+
+class S3TestBundle:
+    """
+    A test bundle staged in S3
+
+    This class does a little bit of "double duty" as we also use it to store the uuid and versions used with the API
+    """
+    BUCKET_TEST_FIXTURES = get_env('DSS_S3_BUCKET_TEST_FIXTURES')
+
+    def __init__(self, path, bucket=BUCKET_TEST_FIXTURES):
+        self.bucket = boto3.resource('s3').Bucket(bucket)
+        self.path = path
+        self.files = self.enumerate_bundle_files()
+        self.uuid = str(uuid.uuid4())
+        self.version = None
+
+    def enumerate_bundle_files(self):
+        object_summaries = self.bucket.objects.filter(Prefix=f"{self.path}/")
+        return [S3File(objectSummary, self) for objectSummary in object_summaries]
+
+
+class S3File:
+    """
+    A test file staged in S3
+    """
+
+    def __init__(self, object_summary, bundle):
+        self.bundle = bundle
+        self.path = object_summary.key
+        self.metadata = object_summary.Object().metadata
+        self.indexed = True if self.metadata['hca-dss-content-type'] == "application/json" else False
+        self.name = os.path.basename(self.path)
+        self.url = f"s3://{bundle.bucket.name}/{self.path}"
+        self.uuid = str(uuid.uuid4())
+        self.version = None
+
+
+class StorageTestSupport(DSSAsserts):
```

Comment:
Yes, initially I also thought this would be a set of test functions, so I'll review how that evolved to the current code. The first issue was the test support code uses the unittest assert* methods, typically obtained by inheriting from unittest.TestCase. If these were just test functions outside of a class, I didn't know of a clean way for the test functions to call assert*. If you know of a clean way to do that please let me know. But that is the main reason this test support code is in a class. It is also a cohesive collection of test support operations that work well together, so grouping them in a class seemed reasonable.
Once they were in a class, I did not originally inherit from DSSAsserts, and just included the StorageTestSupport in the list of classes that were inherited by the tests that needed that support. This worked fine (tests passed), but mypy complained:
$ make test
flake8 dss tests chalice/*.py daemons/*/*.py
mypy --ignore-missing-imports dss tests
tests/infra.py:141: error: "StorageTestSupport" has no attribute "assertPutResponse"
tests/infra.py:151: error: "StorageTestSupport" has no attribute "assertIs"
tests/infra.py:152: error: "StorageTestSupport" has no attribute "assertIn"
tests/infra.py:156: error: "StorageTestSupport" has no attribute "assertPutResponse"
tests/infra.py:163: error: "StorageTestSupport" has no attribute "assertIs"
tests/infra.py:164: error: "StorageTestSupport" has no attribute "assertIn"
tests/infra.py:186: error: "StorageTestSupport" has no attribute "assertGetResponse"
tests/infra.py:196: error: "StorageTestSupport" has no attribute "assertEqual"
tests/infra.py:201: error: "StorageTestSupport" has no attribute "fail"
tests/infra.py:202: error: "StorageTestSupport" has no attribute "assertEqual"
tests/infra.py:203: error: "StorageTestSupport" has no attribute "assertEqual"
tests/infra.py:204: error: "StorageTestSupport" has no attribute "assertEqual"
tests/infra.py:208: error: "StorageTestSupport" has no attribute "assertGetResponse"
tests/infra.py:213: error: "StorageTestSupport" has no attribute "assertEqual"
tests/infra.py:214: error: "StorageTestSupport" has no attribute "assertEqual"
make: *** [mypy] Error 1
I addressed this by changing StorageTestSupport to extend DSSAsserts.
Is there a better way?